redux/redux-reducers.html

```diff
@@ -11,36 +27,110 @@
     (function() {
 
 
-        function user(
+        function identity(
             state = {
                 user: [],
-                loggedIn: false
+                loggedIn: false,
+                token: ""
             },
             action
         ) {
             switch (action.type) {
                 case LOGOUT_USER:
                     return Object.assign({}, state, {
                         loggedIn: false,
-                        user: []
+                        user: [],
+                        token: ""
                     });
                 case LOGIN_USER:
                     return Object.assign({}, state, {
-                        loggedIn: true,
+                        token : action.token,
+                        lastUpdated: action.receivedAt
                     });
                 case RECEIVE_USER:
                     return Object.assign({}, state, {
                         user: action.user,
+                        loggedIn: true,
+                        lastUpdated: action.receivedAt
+                    });
+
+                case CLEAR_TOKEN:
+                    return Object.assign({}, state, {
+                        token: "",
                         lastUpdated: action.receivedAt
                     });
                 default:
                     return state
             }
         }
 
+        function organisation(
+            state = {
+                id: 0,
+                profile: [],
+                clientId: "",
+                default: 0
+            },
+            action
+        ) {
+            switch (action.type) {
+                case RECEIVE_ORGANISATION:
+                    return Object.assign({}, state, {
+                        id : action.id,
+                        profile: action.profile,
+                        lastUpdated: action.receivedAt
+                    });
+                case CLEAR_ORGANISATION:
+                    return Object.assign({}, state, {
+                        id : 0,
+                        profile: [],
+                        clientId: ""
+                    });
+                case RECEIVE_CLIENT_ID:
+                    return Object.assign({}, state, {
+                        clientId: action.clientId,
+                        lastUpdated: action.receivedAt
+                    });
+                default:
+                    return state
```

Comment:
Will be usefull for further debuggin if we but log message here, if  action type is not found.